Update to use remark-mermaidjs (#19)
ChappIO/gatsby-remark-mermaid: README.md

# gatsby-remark-mermaid
[![npm](https://img.shields.io/npm/v/gatsby-remark-mermaid.svg?style=flat-square)](https://www.npmjs.com/package/gatsby-remark-mermaid)

> Creates [mermaid](https://mermaidjs.github.io/) graphs and diagrams in your markdown files.

Reviewer: I prefer an imperative style tag line.

```suggestion
Create [mermaid](https://mermaidjs.github.io/) graphs and diagrams in your markdown files.
```

> This plugin uses [remark-mermaidjs](https://github.com/remcohaszing/remark-mermaidjs) to generate SVG diagrams at build-time. The mermaid code blocks are replaced with an inline SVGs in the generated HTML. This prevents any runtime

Reviewer: ```suggestion
This plugin uses [remark-mermaidjs](https://github.com/remcohaszing/remark-mermaidjs) to generate SVG diagrams at build time. The mermaid code blocks are replaced with an inline SVG in the generated HTML. This prevents any runtime
```

dependencies on `mermaid.js`.

## Install

> ```bash

Reviewer: ```suggestion
```sh
```

npm install gatsby-remark-mermaid gatsby-transformer-remark
```

> ## How to use

Reviewer: Thank you for structuring the readme!

I’ve been trying to use a more consistent readme layout across my repositories, including consistent headers.

```suggestion
## Usage
```

Configure this plugin as a plugin of [gatsby-transformer-remark](https://www.gatsbyjs.com/plugins/gatsby-transformer-remark/).

**NOTE:** Make sure you add this plugin **before** any other plugins that process code blocks.

Example configuration

```js
// In your gatsby-config.js

> plugins: [

Reviewer: Let’s provide fully valid JavaScript, including:

```js
module.exports = {
  // …
}
```

  {
    resolve: 'gatsby-transformer-remark',
    options: {
      plugins: [
        {
          resolve: `gatsby-remark-mermaid`,
          options: {
            launchOptions: {
              executablePath: 'path/to/chrome/executable',
            },
            svgo: {
              plugins: [
                { name: 'removeTitle', active: false },
              ],
            },
            mermaidOptions: {
              theme: 'neutral',
              themeCSS: '.node rect { fill: #fff; }',
            },
          },
        },
      ]
    }
  }
],
```

> ## Configuration Options

Reviewer: ```suggestion
## Options
```

The configuration options for this plugin are the same as for `remark-mermaidjs` [provided here](https://github.com/remcohaszing/remark-mermaidjs#options).
However, the table below describes the configuration details as they apply to Gatsby.

| Name                            | Default              | Description                                                                                                                                                                                                                                                               |
| :------------------------------ | :------------------- | :------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |

> | `launchOptions.executablePath`  | `""`                 | (Required) String path to the chrome executable that puppeteer uses to render the mermaid diagrams to SVGs.                                                                                                                                                               |

Reviewer: This option has no default value.

```suggestion
| `launchOptions.executablePath`  |                      | (Required) String path to the chrome executable that puppeteer uses to render the mermaid diagrams to SVGs.                                                                                                                                                               |
```

| `mermaidOptions.theme`          | `"default"`          | (Optional) Name of mermaid theme. Valid options: `'default'`, `'forest'`, `'dark'`, `'neutral'`, `'null'`. See [mermaid themes](https://mermaid-js.github.io/mermaid/#/Setup?id=theme).                                                                                   |
| `mermaidOptions.themeCSS`       | `""`                 | (Optional) Override mermaid styles using `themeCSS`. See [mermaid themes](https://mermaid-js.github.io/mermaid/#/Setup?id=theme).                                                                                                                                         |
| `mermaidOptions.themeVariables` | `{}`                 | (Optional) Override mermaid variables using `themeVariables`. See [Customizing Themes with themeVariables](https://mermaid-js.github.io/mermaid/#/./theming?id=customizing-themes-with-themevariablesSpecifies).)                                                         |

> | `svgo.plugins`                  | `defaultSVGOOptions` | (Optional) Override default optimizations for the generated SVG files. Set to `null` to disable minifying using SVGO completely. See [defaultSVGOOptions](https://github.com/remcohaszing/remark-mermaidjs/blob/529e44babf07b593489efbe7c238a39089066faa/index.ts#L18)).) |

Reviewer: ```suggestion
| `svgo.plugins`                  | `defaultSVGOOptions` | (Optional) Override default optimizations for the generated SVG files. Set to `null` to disable minifying using SVGO completely. See [defaultSVGOOptions](https://github.com/remcohaszing/remark-mermaidjs/blob/v4.0.0/index.ts#L18)).) |
```
